.github/workflows/unit_tests.yml

```diff
@@ -11,7 +11,7 @@ jobs:
           - 3.0.4
           - 3.1.2
           - 3.2.0
-          - git+https://github.com/sphinx-doc/sphinx.git@3.2.x
+          - git+https://github.com/sphinx-doc/sphinx.git@3.3.x
```

Comment:
@utzig Seems like 3.3.x is really broken in that case. With the current setup with requirements jobs run for 3.3.x, install 3.3.x from source, then uninstall it and fetch 3.2.x to satisfy the requirements. This results in a misleading test job running for 3.3.x.

I think commenting the 3.3.x branch out with a comment, like the master branch for Sphinx 4, is a good solution for now. Can you push a fixup for that? I will merge after, the rest is good.

Reply:
Done. I was testing a little bit why it is broken and wanted to be sure the issue is not Breathe related. Turns out some of the fails are due to asserts for valid lineno, which when Sphinx calls `self.get_source_info()[1]` it goes through docutils into `get_source_and_line` in the `MockStateMachine`. So I presume it is possible to have a lineno that is `None` but the C/CPP domain parsers are currently not allowing for it (after a recent change).

Reply:
For the mock I suggest just returning some arbitrary number.

Reply:
Done, it passes the tests, but I am not sure that means we should drop the other commit. TBH for doing another release I would rather have it not depending on Sphinx<3.3, because on one of the projects I work on we already upgrade to 3.3.0! But it is failing for other projects, so maybe for a next release we just wait before 3.3.1? Is there a timeline for that @jakobandersen? Should I drop the requirements @vermeeren?